javaparser-core/src/main/java/com/github/javaparser/ast/CompilationUnit.java

```diff
@@ -99,6 +100,9 @@
 
     @InternalProperty
     private Storage storage;
+    
+    // printer used to print the node
+    private Printable printer;
```

Comment:
Consider putting this in the "data" map to avoid littering the class with fields.

Reply:
It seems to me that semantically an attribute can make sense and clearly indicate an intention otherwise it would be a questioning of the object-oriented paradigm. But if using a map has a consensus on the project I see no objection.

Reply:
Property 'printer' is moved to 'data' map.